guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc

```diff
@@ -0,0 +1,497 @@
+for the API, while all the logic needs to be implemented in a different class (that inherits the API abstract class).
+This way your changes won't be overwritten by generation, but API will be updated.
+=========
+
+After running OpenAPI generator CLI should output information about generated files.
+Now you can open the folder in your favorite IDE or text editor.
+
+You should see the following folder structure:
+
+[source,text]
+----
+./
+├── docs
+│   └── ... # <1>
+├── src/
+│   ├── main/
+│   │   ├── java/
+│   │   │   └── example/micronaut/library
+│   │   │       ├── Application.java # <2>
+│   │   │       ├── controller
+│   │   │       │   └── BooksController.java # <3>
+│   │   │       └── model
+│   │   │           ├── BookAvailability.java # <4>
+│   │   │           └── BookInfo.java
+│   │   └── resources/
+│   │       ├── application.yml # <5>
+│   │       └── logback.xml
+│   └── test/
+│       └── java/
+│           └── example/micronaut/library # <6>
+│               ├── controller
+│               │   └── BooksControllerTest.java
+│               └── model
+│                   ├── BookAvailabilityTest.java
+│                   └── BookInfoTest.java
+├── README.md
+└── ...
+----
+<1> The `docs/` folder contains automatically generated markdown files with documentation about your API.
+<2> The `Application.java` will start the Micronaut server with all the detected Controllers.
+<3> The `BooksController.java` is generated based on paths with `books` tag. It is generated in the package we specified
+for controllers earlier.
+<4> 2 files are generated in the `models/` folder based on schemas we provided in the definition.
+<5> Config file for Micronaut is generated with a default value for server port and other parameters.
+<6> Tests are generated for all the controllers and models.
+
+== Writing the Controller Logic
+
+If you look inside the generated `BookInfo.java` file, you can see the class that was generated with all the parameters
+based on our definition. Notice that the constructor signature has 2 parameters, which were defined as `required` in the
+YAML definition file:
+
+[source,java]
+----
+    public BookInfo(String name, BookAvailability availability);
+----
+
+Along with that it has getters and setters for parameters and jackson serialization annotation.
+
+To add the required functionality to the server we will first create a service to manage books in our library.
+Create file `src/main/java/example/micronaut/library/BookService.java` with the following content:
+
+source:library/BookService[]
+callout:singleton[1]
+
+Now open the `BooksController.java`. The class `BooksController` should have 2 methods named the same
+as the operations we created in the definition file. The method have Micronaut framework annotations describing the
+required API. We will now write their bodies.
+
+First, import and inject a `BookService` instance as a class attribute:
```

Comment:
"import and inject a `BookService` instance as a class attribute:" -> "inject `BookService`:"